File: Backend/API/SSC.GooseTap.Business/Responces/CheckpointResponse.cs
using System;

namespace SSC.GooseTap.Business.Responces
{
    public class CheckpointResponse
    {
        public long Balance { get; set; }
        public int Energy { get; set; }
        public int MaxEnergy { get; set; }
        public decimal ProfitPerHour { get; set; }
        public decimal OfflineIncome { get; set; }
        public DateTime LastSyncDate { get; set; }

        public int Level { get; set; }
        public string Rank { get; set; } = string.Empty;

        // Boosters info
        public int MultitapLevel { get; set; }
        public int EnergyLimitLevel { get; set; }
        public int RechargeSpeedLevel { get; set; }

        public int ProfitPerClick { get; set; }
        public int EnergyRestorePerSecond { get; set; }
    }
}

File: Frontend/lib/api/models/checkpoint_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'checkpoint_response.g.dart';

@JsonSerializable()
class CheckpointResponse {
  final int? balance;
  final int? energy;
  final int? maxEnergy;
  final double? profitPerHour;
  final double? offlineIncome;
  final DateTime? lastSyncDate;
  final int? level;
  final String? rank;

  CheckpointResponse({
    this.balance,
    this.energy,
    this.maxEnergy,
    this.profitPerHour,
    this.offlineIncome,
    this.lastSyncDate,
    this.level,
    this.rank,
  });

  factory CheckpointResponse.fromJson(Map<String, dynamic> json) =>
      _$CheckpointResponseFromJson(json);

  Map<String, dynamic> toJson() => _$CheckpointResponseToJson(this);
}
